app/src/main/java/com/github/onlynotesswent/ui/deck/DeckMenu.kt

```diff
@@ -0,0 +1,521 @@
+
+                // Deck Title
+                Text(
+                    selectedDeck.value!!.name,
+                    style = Typography.displayMedium,
+                    modifier = Modifier.padding(vertical = 10.dp).testTag("deckTitle"))
+                // Deck description
+                Text(
+                    selectedDeck.value!!.description,
+                    style = Typography.bodyMedium,
+                    modifier = Modifier.padding(10.dp).testTag("deckDescription"))
+
+                // Deck play mode buttons
+                FilledTonalButton(
+                    onClick = { playModesShown.value = !playModesShown.value },
+                    modifier = Modifier.padding(vertical = 15.dp).testTag("deckPlayButton")) {
+                      Row(verticalAlignment = Alignment.CenterVertically) {
+                        Text("Play", style = MaterialTheme.typography.headlineMedium)
+                        Icon(Icons.Default.PlayArrow, contentDescription = "play")
+                      }
+                    }
+
+                // Deck sort options
+                SortOptions(sortOptionsShown, sortMode, sortOrder)
+
+                // Deck cards lazy column
+                LazyColumn(
+                    modifier =
+                        Modifier.fillMaxWidth(0.9f)
+                            .testTag("deckFlashcardsList")
+                            .heightIn(max = 600.dp)
+                            .padding(horizontal = 10.dp),
+                    verticalArrangement = Arrangement.spacedBy(10.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally) {
+                      items(sortedFlashcards.value.size) { index ->
+                        FlashcardViewItem(
+                            flashcard = sortedFlashcards.value[index],
+                            deckViewModel = deckViewModel,
+                            flashcardViewModel = flashcardViewModel,
+                            fileViewModel = fileViewModel,
+                            pictureTaker = pictureTaker,
+                            belongsToUser = belongsToUser)
+                      }
+                    }
+              }
+        }
+      }
+}
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Modularised functions for the deck FABs
+
+/**
+ * Composable function that displays a floating action button (FAB) with a dropdown menu for
+ * deck-related actions for public decks. The dropdown menu contains options to save the deck to
+ * favorites and create a local copy of the deck.
+ *
+ * @param expandedDropdownMenu The state for the dropdown menu expansion.
+ * @param onSaveClick The callback to be invoked when the "Save to favorites" menu item is clicked.
+ * @param onSaveCopyClick The callback to be invoked when the "Create local copy" menu item is
+ *   clicked.
+ */
+@Composable
+private fun PublicDeckFab(
```

Comment:
Very good to even document private function, it help when reading the code

Reply:
you're welcome